lib/c_glib/src/thrift/c_glib/protocol/thrift_compact_protocol.c

```diff
@@ -120,7 +120,11 @@ thrift_bitwise_cast_gdouble (const guint64 v)
 static guint64
 i64_to_zigzag (const gint64 l)
 {
-  return (l << 1) ^ (l >> 63);
```

Comment:
Couldn't we put ULL qualifiers on everything here to force them to unsigned long long types?